python/pyarrow/_compute.pyx

```diff
@@ -316,3 +322,58 @@ cdef class FilterOptions(FunctionOptions):
 
     cdef const CFunctionOptions* options(self) except NULL:
         return &self.filter_options
+
+
+def cast(object arr, object target_type, bint safe=True):
+    """
+    Cast array values to another data type.
+
+    Example
+    -------
+
+    >>> from datetime import datetime
+    >>> import pyarrow as pa
+    >>> arr = pa.array([datetime(2010, 1, 1), datetime(2015, 1, 1)])
+    >>> arr.type
+    TimestampType(timestamp[us])
+
+    You can use ``pyarrow.DataType`` objects to specify the target type:
+
+    >>> arr.cast(pa.timestamp('ms'))
+    <pyarrow.lib.TimestampArray object at 0x10420eb88>
+    [
+      1262304000000,
+      1420070400000
```

Comment:
Minor note, but for me (on master), this gives actual string date representation instead of milliseconds:

```
In [6]: arr.cast(pa.timestamp('ms'))
Out[6]: 
<pyarrow.lib.TimestampArray object at 0x7f95c16bcc48>
[
  2010-01-01 00:00:00.000,
  2015-01-01 00:00:00.000
]
```

Reply:
Ah, I see now it is just copied from the old docstrings below. That explains why the old repr is used here ;)